corehq/apps/reports/tasks.py

```diff
@@ -257,14 +259,45 @@ def build_form_multimedia_zip(
 
     with TransientTempfile() as temp_path:
         with open(temp_path, 'wb') as f:
-            _write_attachments_to_file(temp_path, num_forms, forms_info, case_id_to_name)
+            _write_attachments_to_file(temp_path, num_forms, forms_info, case_id_to_name,
+                                       build_form_multimedia_zip)
         with open(temp_path, 'rb') as f:
             zip_name = 'multimedia-{}'.format(unidecode(export.name))
             _save_and_expose_zip(f, zip_name, domain, download_id, owner_id)
 
     DownloadBase.set_progress(build_form_multimedia_zip, num_forms, num_forms)
 
 
+@task(serializer='pickle')
+def build_form_multimedia_zipfile(
```

Comment:
Deprecates `build_form_multimedia_zip`.
Difference being calling `get_form_ids_with_multimedia` instead of `get_form_ids_having_multimedia` which now accepts `es_query`.
Plus, `_write_attachments_to_file` is called with the new method name to set progress correctly.

Reply:
Same.